### ISSUE TITLE
Adding a new custom model (via api/provider) is not persisted

### ISSUE THREAD

**Issue by marcardar** [ankidroid/Anki-Android#4112]
Steps (DO NOT switch to AnkiDroid app at any point):

1. Add a new model using AddContentApi#addNewCustomModel()
2. Add a new note (to a deck) based on this model, using AddContentApi#addNewNote()
3. Observe that AddContentApi.findModelIdByName(model.getName(), fields.length) is non-null (as expected)
4. Force Stop AnkiDroid
5. Observe that AddContentApi.findModelIdByName(model.getName(), fields.length) is now null

Note: there seems to be no such issue with Decks even though there is also no Decks#flush call in the equivalent place.

This can be fixed by calling Models.flush() in the appropriate place in the provider's insert() implementation.  This is buried in #4110 .

However, I'm wondering if there is any way for the API to trigger that sync() on existing versions of the provider currently out there in the wild?

**Comment by marcardar** [comment]
Here is an example of an exception in AnkiDroid when going to the Card browser after a model was not persisted properly:

```
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'org.json.JSONArray org.json.JSONObject.getJSONArray(java.lang.String)' on a null object reference
at com.ichi2.libanki.Models.fieldMap(Models.java:476)
at com.ichi2.libanki.Note.load(Note.java:108)
at com.ichi2.libanki.Note.<init>(Note.java:70)
at com.ichi2.libanki.Note.<init>(Note.java:56)
at com.ichi2.libanki.Collection.getNote(Collection.java:423)
at com.ichi2.libanki.Card.note(Card.java:319)
at com.ichi2.libanki.Card._getQA(Card.java:285)
at com.ichi2.anki.CardBrowser.updateSearchItemQA(CardBrowser.java:930)
at com.ichi2.async.DeckTask.doInBackgroundRenderBrowserQA(DeckTask.java:646)
at com.ichi2.async.DeckTask.doInBackground(DeckTask.java:301)
at com.ichi2.async.DeckTask.doInBackground(DeckTask.java:66) 
at android.os.AsyncTask$2.call(AsyncTask.java:295) 
at java.util.concurrent.FutureTask.run(FutureTask.java:237) 
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234) 
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) 
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) 
at java.lang.Thread.run(Thread.java:818) 
```

**Comment by marcardar** [comment]
This was fixed in #4110
